lxd/db_test.go

```diff
@@ -7,40 +7,45 @@ import (
 	"testing"
 )
 
-const CONTAINER_AND_PROFILE string = `
+const DB_FIXTURES string = `
     INSERT INTO containers (name, architecture, type) VALUES ('thename', 1, 1);
     INSERT INTO profiles (name) VALUES ('theprofile');
     INSERT INTO containers_profiles (container_id, profile_id) VALUES (1, 2);
     INSERT INTO containers_config (container_id, key, value) VALUES (1, 'thekey', 'thevalue');
     INSERT INTO containers_devices (container_id, name) VALUES (1, 'somename');
-    INSERT INTO containers_devices_config (key, value, container_device_id) VALUES ('configkey', 'configvalue', 1);`
-
-const IMAGE string = `
+    INSERT INTO containers_devices_config (key, value, container_device_id) VALUES ('configkey', 'configvalue', 1);
     INSERT INTO images (fingerprint, filename, size, architecture, creation_date, expiry_date, upload_date) VALUES ('fingerprint', 'filename', 1024, 0,  1431547174,  1431547175,  1431547176);
     INSERT INTO images_aliases (name, image_id, description) VALUES ('somealias', 1, 'some description');
-    INSERT INTO images_properties (image_id, type, key, value) VALUES (1, 0, 'thekey', 'some value');`
+    INSERT INTO images_properties (image_id, type, key, value) VALUES (1, 0, 'thekey', 'some value');
+    INSERT INTO profiles_config (profile_id, key, value) VALUES (2, 'thekey', 'thevalue');`
 
-func Test_deleting_a_container_cascades_on_related_tables(t *testing.T) {
-	var db *sql.DB
+//  This Helper will initialize a test in-memory DB.
+func createTestDb(t *testing.T) (db *sql.DB) {
 	var err error
-	var count int
-	var statements string
-
 	db, err = initializeDbObject(":memory:")
-	defer db.Close()
 
 	if err != nil {
-		t.Error(err)
+		t.Fatal(err)
 	}
 
-	// Insert a container and a related profile.
-	_, err = db.Exec(CONTAINER_AND_PROFILE)
+	_, err = db.Exec(DB_FIXTURES)
 	if err != nil {
-		t.Error(err)
+		t.Fatal(err)
 	}
+	return // db is a named output param
+}
+
+func Test_deleting_a_container_cascades_on_related_tables(t *testing.T) {
+	var db *sql.DB
+	var err error
+	var count int
+
+	// Insert a container and a related profile.
+	db = createTestDb(t)
+	defer db.Close()
 
 	// Drop the container we just created.
-	statements = `DELETE FROM containers WHERE name = 'thename';`
+	statements := `DELETE FROM containers WHERE name = 'thename';`
```

Comment:
Not in the tests, surely?

This creates the condition in which the containers should cascade delete
the related entries (to check that it actually works)

Or... Or I don't understand what you mean :)
On 14 May 2015 21:36, "Tycho Andersen" <notifications@github.com> wrote:

> In lxd/db_test.go
> <https://github.com/lxc/lxd/pull/661#discussion_r30354713>:
>
> >
> >  	// Drop the container we just created.
> > -	statements = `DELETE FROM containers WHERE name = 'thename';`
> > +	statements := `DELETE FROM containers WHERE name = 'thename';`
>
> I'm confused, didn't we just drop these in a previous patch?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lxc/lxd/pull/661/files#r30354713>.
>